### ISSUE TITLE
Can only have one tech repair a dropship at a time

### ISSUE THREAD

**Issue by BLOODWOLF333** [MegaMek/mekhq#153]
i have a Manatee (standard) with two vessel techs (hire full compliment) and the repair tab only list one tech as available to work on the dropship after it has been damaged.  Tried deleting both the vessel techs and re-hiring full compliment, concern still present.  Unassigned and reassigned them to the DS, GM removed their vessel tech skills then added it back.........

![example](https://cloud.githubusercontent.com/assets/16950832/18259466/cd8cc5d0-73a9-11e6-95cc-514e8b854e94.png)

![github](https://cloud.githubusercontent.com/assets/16950832/18259468/d25fe9c0-73a9-11e6-9f45-16b345efa69c.png)

[vesseltechbug.zip](https://github.com/MegaMek/mekhq/files/455799/vesseltechbug.zip)

**Comment by pheonixstorm** [comment]
Not a bug.  The assigned tech represents all the techs aboard a ship as they are considered the "crew" as far as repairs are concerned. This comes from Strat Ops. Don't recall the page number.

**Comment by BLOODWOLF333** [comment]
..........why not at least just list them all?  I know i know i need to buy that book but, thats not intuitive visually at all.

**Comment by pheonixstorm** [comment]
SO just tells us the "Crew" handles all the work so its not just one person to choose from but the crew, which is an average of the tech skill. HQ though does need to pick someone in order to function properly

**Comment by mkerensky** [comment]
Do they all get XP if such is turned on for successful skill rolls?  Does HQ average the skills of the various crewmembers?  If the crew is being displayed this way, can we have it renamed "Dropship XXX Crew" to be a little more intuitive?

**Comment by pheonixstorm** [comment]
For the XP don't know, never really looked into it. Yes it does average the skill for tech, gunners, pilots (or it is supposed to). As for the names... unlikely. Something to ask ralgith about but I don't see it getting changed to something like this as it would require a fair amount of work to do I think.

**Comment by Dylan-M** [comment]
I'm going to close this one as never happening. @Akjosch is refactoring the entire repair system.

**Comment by mkerensky** [comment]
I did confirm that only one Dropship crewman gets the XP for successful skill rolls. It might be a random person on the crew - didn't seem to be only the named crewman representing the average.

**Comment by Dylan-M** [comment]
I'm not convinced we're handling this correctly anyway. For example, we're only giving 480 minutes, but you have X crewmen * 480 minutes work-day. So, if you have a 3 man tech team aboard ship... then you have 1440 minutes of tech time per day available. This would jive much better with the extremely _long_ repair times on DS, JS, and WS. ASF OTOH, are just a cluster f...

**Comment by HammerGS** [comment]
I agree with Dylan-M, all the crew should be working on the repairs.

**Comment by mkerensky** [comment]
Yeah I've noticed that the repair times are painful, and you pretty much have to use extra time with a green or regular crew, because if you fail a roll it's impossible to repair the item until the crew levels up.  I've only fought one major space battle using MHQ, but most of my dropships look like they're going to be down for about 12-16 weeks making repairs.

It may deserve a separate ticket, but I did also notice in this process that if you salvage a dropship component (using a tech with the tech/vessel skill), you can't repair it in the warehouse - says you don't have any techs with the correct skillset. This might have been fixed with the recent change that assigned customization to the virtual engineer instead of the commander.

**Comment by pheonixstorm** [comment]
A ships crew is worth 480 minutes just like everyone else. SO 168/169 under the crew entry. The example shows a quarter being 120 minutes. Don't know if there is a spot that says 480, but that section does clarify a bit. Also near there it says:

> When docked at a fully equipped space yard, yardship, or port the time required to perform maintenance and repairs is reduced to half.

**Comment by HammerGS** [comment]
The following errata 'could' have an impact on this conversation.

http://bg.battletech.com/forums/index.php?topic=54187.0

In the sense of one unit helping another with repair.

**Comment by mkerensky** [comment]
Another issue with DropShip repairs that I'm finding in 0.41.25 - Repair crews seem to count as Shorthanded and get a penalty if they don't have a full complement, including Pilots and Gunners. You also don't have a repair crew at all that can salvage/repair components unless you have at least 1 Gunner and 1 Pilot in addition to Crewmembers.  See the attached save.  My repair bay is kind of full, but I have 5 Tiamat dropships - 1 to repair and 4 at salvage quality.  I have two crews busy salvaging parts from the 4 to put the 1 back together, and they have the shorthanded issue. It's the same crews that tried to salvage thrusters where I found I couldn't repair the damaged parts in the warehouse too. 
[DropShip Repairs.zip](https://github.com/MegaMek/mekhq/files/556849/DropShip.Repairs.zip)

**Comment by mkerensky** [comment]
This one can be assigned to me at this point. 
It's actually a whole bunch of separate issues that I'm going to wind up resolving with different PRs, but for right now I've improved the functionality of the ship's engineer virtual crewmember via edge and XP distribution to the component crewmen.

**Comment by BLOODWOLF333** [comment]
1.  I think a vessel crew regardless of how many personnel make up the crew in HQ should equal 480 minutes like every other crew.
2.  The entire crew works on the repairs for however many hours/days/months so every member should receive the same amount of xp once the repair is complete and call it done.

Right now I think it is just using the highest skilled crewmember?  So that green crewmember is useless then and doesnt need to be there.  They will never gain xp and increase in rank so, idk maybe having multiple crewmembers should also give us more time each day?  Or at least just give them all xp for the job that they worked on is my two cents.

**Comment by sixlettervariables** [comment]
@mkerensky is this still an issue as of the 0.45.X line? I'm going to tentatively close this as code around this has changed considerably over the past year or so.

**Comment by mkerensky** [comment]
Yeah this is fixed and has been for a while. Sorry I forgot to update it way back when.
